File: console/src/types/betlog.ts
export interface BetLogSummary {
  bets: number
  settled_bets: number
  open_bets: number
  stake_usd: number
  pnl_usd: number
  roi: number
  mean_clv: number
  hit_rate: number
}

export interface BetLogRow {
  timestamp: string
  id: string
  market: string
  model_market: string
  edge: number
  size: number
  stake_usd: number
  mode: string
  state: string
  result: string
  clv: number | null
  strategy: string
}

export interface BetLogResponse {
  summary: BetLogSummary
  rows: BetLogRow[]
}

File: api/models/betlog.py
from __future__ import annotations

from pydantic import BaseModel


class BetLogSummary(BaseModel):
    bets: int
    settled_bets: int
    open_bets: int
    stake_usd: float
    pnl_usd: float
    roi: float
    mean_clv: float
    hit_rate: float


class BetLogRow(BaseModel):
    timestamp: str
    id: str
    market: str
    model_market: str
    edge: float
    size: float
    stake_usd: float
    mode: str
    state: str
    result: str
    clv: float | None
    strategy: str


class BetLogResponse(BaseModel):
    summary: BetLogSummary
    rows: list[BetLogRow]
